app/src/main/java/com/example/poste/PosteApplication.java

```diff
@@ -23,15 +33,75 @@ public class PosteApplication extends Application {
+     */
+    public void dataMockUp() {
+        Post.Builder builder1 = new Post.Builder();
+        Post.Builder builder2 = new Post.Builder();
+        Post.Builder builder3 = new Post.Builder();
+        builder1.setTitle("Look at these ones!");
+        builder2.setTitle("Look at these twos!");
+        builder3.setTitle("Wowza! cool threes!");
+        builder1.setDescription("Desc 1");
+        builder2.setDescription("Desc 2");
+        builder3.setDescription("Desc 3");
+        builder1.setUrl("www.youtube.com");
+        builder2.setUrl("www.facebook.com");
+        builder3.setUrl("www.https://github.com/.com");
+        builder1.setId("a");
+        builder2.setId("b");
+        builder3.setId("c");
+        Post post1 = builder1.build();
+        Post post2 = builder2.build();
+        Post post3 = builder3.build();
+        Folder.Builder fBuild1 = new Folder.Builder();
+        Folder.Builder fBuild2 = new Folder.Builder();
+        fBuild1.setId("fa");
+        fBuild2.setId("fb");
+        fBuild1.setTitle("Folder 1");
+        fBuild2.setTitle("Folder 2");
+        fBuild1.setUserPermission("full_access");
+        fBuild2.setUserPermission("viewer");
+        ArrayList<Post> mockList1 = new ArrayList<>();
+        ArrayList<Post> mockList2 = new ArrayList<>();
+        mockList1.add(post1);
+        mockList1.add(post2);
+        mockList2.add(post3);
+        fBuild1.setPosts(mockList1);
+        fBuild2.setPosts(mockList2);
+        Folder folder1 = fBuild1.build();
+        Folder folder2 = fBuild1.build();
+        com.example.poste.models.User.getUser().addFolder(folder1);
+        com.example.poste.models.User.getUser().addFolder(folder2);
+        selectedFolder = com.example.poste.models.User.getUser().getFolders().get(0);
+    }
+
+    /**
+     * Debug code (remove this function in final build)
+     */
+    public void printMockData(){
+        List<Folder> usersMockFolders = com.example.poste.models.User.getUser().getFolders();
+        System.out.println("The user has the folders shown below:");
+        for (int i = 0; i < usersMockFolders.size(); i++) {
+            Folder f = usersMockFolders.get(i);
+            System.out.println("Folder Id = " + f.getId() + " --- Title = " + f.getTitle() + " --- Access Level = " + f.getUserPermission() + " And holds these posts:");
+            List<Post> mps = f.getPosts();
+            for (int k = 0; k < mps.size(); k++) {
+                Post p = mps.get(k);
+                System.out.println("     Post Id = " + p.getId() + " --- Title = " + p.getTitle() + " --- Description = " + p.getDescription() + " --- Link = " + p.getUrl());
+            }
+        }
     }
 
     /**
      * Get the currently logged-in user
      *
      * @return The User object representing the currently logged-in user
      */
-    public static User getCurrentUser() {
+    public static com.example.poste.api.poste.models.User getCurrentUser() {
```

Comment:
Thanks for changing the model. Ideally, everywhere that calls `getCurrentUser()` should be calling `User.getUser()` (the singleton getter)... until that time, can you have this method return `User.getUser()`? This will, at the very least, ensure we don't somehow use / access / do something with an incorrect User instance.

Reply:
Please be particular in user stories / tasks so that individual needs aren't forgotten.

I'd much rather see this here so we aren't changing the same code twice, and since this code is already changing here -- also, merge conflicts are the result of multiple developers working on the same code, so no merge conflict looks to arise from this. 

I'll mark this as resolved for now, but have updated the Taiga task. Please consider for the future.